sysl2/sysl/visitor.go

```diff
@@ -207,17 +226,21 @@ func (v *SequenceDiagramVisitor) Visit(e Element) error {
 			log.Errorln(err)
 		}
 	}()
-
+	var err error
 	switch t := e.(type) {
 	case *EndpointCollectionElement:
-		return v.visitEndpointCollection(t)
+		err = v.visitEndpointCollection(t)
```

Comment:
Move this after the for loop, and change it to a `return …`.

Reply:
Changed

Reply:
This can't be moved below the for loop, as the bbVal.VisitCount is calculated in the call v.visitEndpointCollection(t).